src/Microsoft.PowerShell.Security/security/CertificateProvider.cs

```diff
@@ -508,19 +457,9 @@ public IntPtr GetNextCert(IntPtr certContext)
             }
             if (Valid)
             {
-                if (_filterHandle != null)
```

Comment:
probably just performance.  Looking at the CmdLet we only supported filtering by codesigning certs which the tests pass after the change.